ui/js/component/header/index.js

```diff
@@ -12,18 +12,23 @@ import {
   doHistoryForward,
 } from "actions/navigation";
 import Header from "./view";
+import { selectIsUpgradeAvailable } from "../../selectors/app";
+import { doDownloadUpgrade } from "../../actions/app";
 
 const select = state => ({
   isBackDisabled: selectIsBackDisabled(state),
   isForwardDisabled: selectIsForwardDisabled(state),
+  isUpgradeAvailable: selectIsUpgradeAvailable(state),
   balance: formatCredits(selectBalance(state) || 0, 1),
   publish: __("Publish"),
+  upgradeLabel: __("Upgrade App"),
```

Comment:
I see you followed the pattern established above, but I'm guessing this is legacy behavior from a time when that label could switch somehow. Both `upgradeLabel` and `publish` can just be put directly in `view.jsx`.